server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/ReprocessingService.java

```diff
@@ -86,12 +86,12 @@ static class Reprocessor implements Closeable {
             this.configuration = configuration;
         }
 
-        private void reprocess(MailRepository repository, Mail mail) {
+        private void reprocess(MailRepository repository, Mail mail, MailKey key) {
```

Comment:
If we have the mail we have the key? IE this method signature can likely stay the same.